services/workflows-service/src/transaction/transaction.controller.external.ts

```diff
@@ -24,38 +35,62 @@ export class TransactionControllerExternal {
 
   @Post()
   @UseCustomerAuthGuard()
-  @swagger.ApiCreatedResponse({ type: TransactionCreateDto })
+  @swagger.ApiCreatedResponse({ type: TransactionCreatedDto })
   @swagger.ApiForbiddenResponse()
   async create(
-    @Body() body: TransactionCreateDto,
+    @Body(new ValidationPipe()) body: TransactionCreateDto,
+    @Res() res: express.Response,
     @CurrentProject() currentProjectId: types.TProjectId,
   ) {
-    this.logger.log('create transaction');
-    const transaction: TransactionCreateDto = {
-      ...body,
+    const creationResponses = await this.service.createBulk({
+      transactionsPayload: [body],
       projectId: currentProjectId,
-    };
+    });
+    const creationResponse = creationResponses[0]!;
 
-    const createdTransaction = await this.service.create(transaction);
+    if ('error' in creationResponse) {
+      throw creationResponse.error;
+    }
 
-    return createdTransaction;
+    res.status(201).json(creationResponse);
   }
 
-  @Post('/batch')
+  @Post('/bulk')
   @UseCustomerAuthGuard()
-  @swagger.ApiCreatedResponse({ type: [TransactionCreateDto] })
+  @swagger.ApiCreatedResponse({ type: BulkTransactionsCreatedDto })
   @swagger.ApiForbiddenResponse()
-  async createBatch(
-    @Body() body: TransactionCreateDto[],
-    @Res() response: express.Response,
+  async createBulk(
+    @Body(new ParseArrayPipe({ items: TransactionCreateDto })) body: TransactionCreateDto[],
+    @Res() res: express.Response,
     @CurrentProject() currentProjectId: types.TProjectId,
   ) {
-    const batchCreateResponse = await this.service.createBatch(body);
-    if (batchCreateResponse.overallStatus === 'partial') {
-      response.status(207);
-    }
+    const creationResponses = await this.service.createBulk({
+      transactionsPayload: body,
+      projectId: currentProjectId,
+    });
```

Comment:
we already agreed on bulk response,  which indicates for each row if had an issue,

error response for bulk for example: `Alerts - Bulk assign - wrong alert id` API: `/api/v1/external/alerts/decision` 
```
 
{
    "overallStatus": "failed",
    "response": [
        {
            "alertId": "clshptjjr03pipgdoq7108",
            "status": "failed",
            "error": {
                "message": "Alert not found or not updated."
            }
        }
    ]
}
```

Reply:
Also success should be look like :
```
{
    "overallStatus": "success",
    "response": [
        {
            "alertId": "clshptjjr03pipgdoq7108hip",
            "status": "success"
        }
    ]
}
```